nodeup/pkg/model/context.go

```diff
@@ -598,29 +599,31 @@ func (c *NodeupModelContext) GetPrivateKey(name string) ([]byte, error) {
 
 func (b *NodeupModelContext) AddCNIBinAssets(c *fi.ModelBuilderContext, assetNames []string) error {
 	for _, assetName := range assetNames {
-		if err := b.addCNIBinAsset(c, assetName); err != nil {
+		re, err := regexp.Compile(fmt.Sprintf("^%s$", assetName))
```

Comment:
I'm a little worried about regexp metacharacters, such as ".", in the assetName.

Perhaps use `regexp.Compile(fmt.Sprintf("^%s$", regexp.QuoteMeta(assetName)))` instead?

Reply:
Sounds reasonable. Thanks for the suggestion.